In [6]:
import numpy as np

In [7]:
import scipy.stats as stats

**Задача №1.** Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

 Уровень значимости α = 1 - 0.95 = 0.05



In [12]:
sigma = 16
X = 80
n = 256
alpha = 0.05

Так как известна $\sigma$ генеральной совокупности выбираем Z-критерий

Определим табличный Z-критерий:

In [43]:
Z = round((stats.norm.ppf(alpha/2)), 2)

In [44]:
Z

-1.96

In [20]:
interval_1 = X + Z * (sigma / n**0.5)

In [22]:
interval_1

78.04

In [21]:
interval_2 = X - Z * (sigma / n**0.5)

In [23]:
interval_2

81.96

**Ответ:** Доверительный интервал для математического ожидания с надежностью 0,95 составляет: [78.04; 81.96]

**Задача №2.** В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1. Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

 Уровень значимости α = 1 - 0.95 = 0.05

In [9]:
alpha = 0.05
array=np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [10]:
n = len(array)

In [11]:
n

10

Определим среднее значение выборки:

In [33]:
X = round((np.mean(array)), 2)

In [34]:
X

6.59

Так как не известна $\sigma$ генеральной совокупности выбираем t-критерий (критерий Стьюдента)

Определим стандартное несмещенное отклонение:

In [14]:
sigma_std = round((np.std(array, ddof=1)), 2)

In [15]:
sigma_std

0.45

In [21]:
t=round((stats.t.ppf(1-alpha/2,df=n-1)), 2)

In [22]:
t

2.26

In [35]:
interval_1 = round((X - t * (sigma_std / n**0.5)), 2)


In [36]:
interval_1

6.27

In [38]:
interval_2= round((X + t * (sigma_std / n**0.5)), 2)

In [39]:
interval_2

6.91

**Ответ:** Доверительный интервал, помогающий оценить истинное значение величины Х, покрывающего это значение с доверительной вероятностью 0,95 следующий: [6.27; 6.91]


**Задача №3.** Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170. <br> Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175. <br>
Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

 Уровень значимости α = 1 - 0.95 = 0.05

In [68]:
alpha = 0.05
array_1 = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
array_2 = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])

Определим среднее значение выборки:

In [77]:
X_1 = round((np.mean(array_1)), 2)
print(X_1)


166.2


In [70]:
X_2 = round((np.mean(array_2)), 2)
print(X_2)

168.1


In [71]:
delta = round((X_1 - X_2), 2)
print(delta)

-1.9


In [72]:
n_1 = len(array_1)
print(n_1)

10


In [73]:
n_2 = len(array_2)
print(n_2)

10


Определим несмещенную дисперсию выборки Х:

In [83]:
D_1 = round((np.var(array_1, ddof=1)), 2)
print(D_1)

91.07


In [85]:
D_2 = round((np.var(array_2, ddof=1)), 2)
print(D_2)

60.1


Объединённая оценка дисперсии D (среднее арифметическое дисперсии):

In [93]:
D = round(((D_1+D_2)/2), 2)
print(D)

75.58


Стандартная ошибка разности средних SE:

In [92]:
SE = np.sqrt(D/n_1 + D/n_2)
print(round(SE, 2))

3.89


In [99]:
t=stats.t.ppf(1-alpha/2,df=n_1 + n_2-2)
print(round(t, 2))

2.1


Доверительный интервал:

In [100]:
interval_1 = delta - t*SE
print(round(interval_1, 2))

-10.07


In [101]:
interval_2 = delta + t*SE
print(round(interval_2, 2))

6.27


**Ответ:** 95% доверительный интервал для разности среднего роста родителей и детей следующий: [-10.07, 6.27]